core/network/src/packets.rs

```diff
@@ -1484,13 +1537,56 @@ pub struct ConfirmTransactionClientbound {
     pub accepted: bool,
 }
 
-#[derive(Default, AsAny, Packet, Clone)]
+#[derive(Default, AsAny, Clone)]
 pub struct OpenWindow {
     pub window_id: u8,
     pub window_type: String,
     pub window_title: String, // Chat
     pub number_of_slots: u8,
-    pub entity_id: i32,
+    pub entity_id: Option<i32>,
+}
+
+impl Packet for OpenWindow {
+    fn read_from(&mut self, buf: &mut Cursor<&[u8]>) -> anyhow::Result<()> {
+        self.window_id = buf.try_get_u8()?;
+        self.window_type = buf.try_get_string()?;
+        self.window_title = buf.try_get_string()?;
+        self.number_of_slots = buf.try_get_u8()?;
+
+        if self.window_type == "EntityHorse" {
```

Comment:
This could be written more concisely with

```rust
self.entity_id = if self.window_type == "EntityHorse" { Some(buf.try_get_i32()? } else { None };
```